Mathlib/LinearAlgebra/Multilinear/TensorProduct.lean

```diff
@@ -37,39 +78,19 @@ introduces `Sum.elim N'₁ N'₂` types in the result which are difficult to wor
 to the simple case defined here. See [this zulip thread](
 https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/Instances.20on.20.60sum.2Eelim.20A.20B.20i.60/near/218484619).
 -/
-@[simps apply]
+@[simps! apply]
 def domCoprod (a : MultilinearMap R (fun _ : ι₁ => N) N₁)
     (b : MultilinearMap R (fun _ : ι₂ => N) N₂) :
-    MultilinearMap R (fun _ : ι₁ ⊕ ι₂ => N) (N₁ ⊗[R] N₂) where
-  toFun v := (a fun i => v (Sum.inl i)) ⊗ₜ b fun i => v (Sum.inr i)
-  map_update_add' _ i p q := by
-    letI := (@Sum.inl_injective ι₁ ι₂).decidableEq
-    letI := (@Sum.inr_injective ι₁ ι₂).decidableEq
-    cases i <;> simp [TensorProduct.add_tmul, TensorProduct.tmul_add]
-  map_update_smul' _ i c p := by
-    letI := (@Sum.inl_injective ι₁ ι₂).decidableEq
-    letI := (@Sum.inr_injective ι₁ ι₂).decidableEq
-    cases i <;> simp [TensorProduct.smul_tmul', TensorProduct.tmul_smul]
+    MultilinearMap R (fun _ : ι₁ ⊕ ι₂ => N) (N₁ ⊗[R] N₂) :=
+  domCoprodDep a b
 
 /-- A more bundled version of `MultilinearMap.domCoprod` that maps
 `((ι₁ → N) → N₁) ⊗ ((ι₂ → N) → N₂)` to `(ι₁ ⊕ ι₂ → N) → N₁ ⊗ N₂`. -/
 def domCoprod' :
     MultilinearMap R (fun _ : ι₁ => N) N₁ ⊗[R] MultilinearMap R (fun _ : ι₂ => N) N₂ →ₗ[R]
       MultilinearMap R (fun _ : ι₁ ⊕ ι₂ => N) (N₁ ⊗[R] N₂) :=
   TensorProduct.lift <|
-    LinearMap.mk₂ R domCoprod
-      (fun m₁ m₂ n => by
-        ext
-        simp only [domCoprod_apply, TensorProduct.add_tmul, add_apply])
-      (fun c m n => by
-        ext
-        simp only [domCoprod_apply, TensorProduct.smul_tmul', smul_apply])
-      (fun m n₁ n₂ => by
-        ext
-        simp only [domCoprod_apply, TensorProduct.tmul_add, add_apply])
-      fun c m n => by
-      ext
-      simp only [domCoprod_apply, TensorProduct.tmul_smul, smul_apply]
+    LinearMap.mk₂ R domCoprod (by aesop) (by aesop) (by aesop) (by aesop)
```

Comment:
Is there much speed difference between these?

Reply:
The `aesop` version is little bit slower, but it not super slow.